src/util/color.js

```diff
@@ -0,0 +1,336 @@
+ *     either be a number between 0 and 1 or a percent.  White space may appear
+ *     before and after numbers, and between the number and a percent symbol.
+ *     Commas are not required.  A slash may be used as a separator before the
+ *     alpha value instead of a comma.  The numbers conform to the CSS number
+ *     specification, and can be signed floating-point values, possibly with
+ *     exponents.
+ *   - `hsl(H, S, L)`, `hsl(H, S, L, A)`, `hsla(H, S, L)`, `hsla(H, S, L, A)`
+ *     Hue, saturation, and lightness with optional alpha (opacity).  Hue is a
+ *     number between 0 and 360 and is interpretted as degrees unless an angle
+ *     unit is specified.  CSS units of `deg`, `grad`, `rad`, and `turn` are
+ *     supported.  Saturation and lightness are percentages between 0 and 100
+ *     and *must* be followed by a percent `%` symbol.  The alpha (opacity)
+ *     channel is optional and is specified as with `rgba(R, G, B, A)`.
+ *   - `transparent` Black with 0 opacity.
+ *
+ * When a number, this is an integer of the form `0xrrggbb` or its decimal
+ * equivalent.  For example, 0xff0000 or 16711680 are both solid red.
+ *
+ * See {@link https://developer.mozilla.org/en-US/docs/Web/CSS/color_value} for
+ * more details on CSS color values.
+ *
+ * @typedef geo.geoColor
+ * @type {geo.geoColorObject|string|number}
+ */
+
+/**
+ * A color value represented as an object.  Although opacity can be specified,
+ * it is not always used.
+ *
+ * @typedef {object} geo.geoColorObject
+ * @property {number} r The red intensity on a scale of [0-1].
+ * @property {number} g The green intensity on a scale of [0-1].
+ * @property {number} b The blue intensity on a scale of [0-1].
+ * @property {number} [a] The opacity on a scale of [0-1].  If unspecified and
+ *      used, it should be treated as 1.
+ */
+
+/**
+ * @typedef {object} geo.util.cssColorConversionRecord
+ * @property {string} name The name of the color conversion.
+ * @property {RegEx} regex A regex that, if it matches the color string, will
+ *      cause the process function to be invoked.
+ * @property {function} process A function that takes (`color`, `match`) with
+ *      the original color string and the results of matching the regex using
+ *      the regex's `exec` function.  It outputs a {@link geo.geoColorObject}
+ *      color object or the original color string if there is still a parsing
+ *      failure.
+ */
+
+var m_memoizeConvertColor = {maxCount: 1000, count: 0, memo: {}};
+
+/**
+ * Store the results of convertColor in dictionary for so repeated requests can
+ * be returned quickly.  If the memoization table gets over a certain size,
+ * just reset it.
+ *
+ * @param {geo.geoColor} origColor The origial color specification.
+ * @param {geo.geoColorObject} resultColor The result of the conversion.
+ * @returns {geo.geoColorObject} The `resultColor`.
+ */
+function memoizeConvertColor(origColor, resultColor) {
```

Comment:
should we call it cacheConvertColor essentially using cache instead of memoize, as it may provide consistent naming convention elsewhere, memoize is fine as well.

Reply:
Okay, lets go with memoization, I do not see the distinction but it seems that others may. +1